docs/architecture/blazor-for-web-forms-developers/introduction.md

```diff
@@ -8,6 +8,50 @@ ms.date: 09/11/2019
+
+But having to bridge two different platforms and ecosystems (.NET and JavaScript) comes with a cost. You as the web developer have to become an expert in two parallel worlds with different languages, frameworks, and tools. Code and logic cannot be easily shared between client and server resulting in duplication and engineering overhead. For some, it is also difficult to keep up with the JavaScript ecosystem, which has a history of evolving at breakneck speed, quickly changing preferences for different frontend frameworks (jQuery, Knockout, Angular, React, Vue, etc.) and build tools (grunt, gulp, webpack, etc.). But given JavaScript's browser monopoly there was little choice in the matter. That is until the web community got together and caused a *miracle* to happen!
+
+## WebAssembly to the rescue!
+
+In 2015 the major browser vendors joined forces in a W3C Community Group to create a new open web standard called WebAssembly. WebAssembly is a byte code for the Web. The idea is that if you can compile your code to WebAssembly it can then run on any browser on any platform at near native speed. Initial efforts focused on C/C++ resulting in some pretty dramatic demonstrations of getting native 3D graphics engines running directly in the browser without any plugins. WebAssembly has since been standardized and implemented by all major browsers. Work on running .NET on WebAssembly was announced in late 2017 and is expected to ship in 2020 including support from .NET 5. The ability to run .NET code directly in the browser enables full-stack web development with .NET.
+
+## Blazor: full-stack web development with .NET
+
+But just being able to run .NET code in a browser doesn't give you an end-to-end experience building client-side web apps. That's where Blazor comes in. Blazor is a new client-side web UI framework based on C# instead of JavaScript that can run directly in the browser via WebAssembly. No browser plugins required. Alternatively, Blazor apps can run server-side on .NET Core and handle all user interactions over a real-time connection with the browser. Blazor includes a full UI component model, as well as built-in facilities for forms & validation, dependency injection, client-side routing, layouts, in-browser debugging, and JavaScript interop. Blazor also comes with great tooling support in Visual Studio and Visual Studio Code.
```

Comment:
```suggestion
But just being able to run .NET code in a browser doesn't give you an end-to-end experience building client-side web apps. That's where Blazor comes in. Blazor is a new client-side web UI framework based on C# instead of JavaScript that can run directly in the browser via WebAssembly. No browser plugins are required. Alternatively, Blazor apps can run server-side on .NET Core and handle all user interactions over a real-time connection with the browser. Blazor includes a full UI component model, as well as built-in facilities for forms and validation, dependency injection, client-side routing, layouts, in-browser debugging, and JavaScript interop. Blazor also comes with great tooling support in Visual Studio and Visual Studio Code.
```